src/components/text-editor/text-editor.tsx

```diff
@@ -142,6 +150,15 @@ export class TextEditor implements FormComponent<string> {
     @Event()
     public change: EventEmitter<string>;
 
+    @Event()
+    public triggerStart: EventEmitter<TextEditorInserter>;
+
+    @Event()
+    public triggerStop: EventEmitter<void>;
+
+    @Event()
+    public triggerChange: EventEmitter<string>;
```

Comment:
I think these should be a bit more symmetric, now all of them have different event details. All of them should probably include the `TextEditor` somehow, maybe we need to wrap it in something like this
```ts
interface TriggerEventDetail {
    /**
     * The trigger that triggered this event
     */
    trigger: string;

    /**
     * The text editor
     */
    textEditor: TextEditor;

    /**
     * Current value of the trigger
     */
    value: string
}
```

`trigger` is needed in case you have multiple trigger characters, otherwise you would not know what trigger emitted the event.

Reply:
Yes! I like this a lot.

Reply:
Fixed in latest commit